src/vsr/client.zig

```diff
@@ -371,7 +371,7 @@ pub fn Client(comptime StateMachine_: type, comptime MessageBus: type) type {
             }
 
             if (inflight.message.header.operation != .register) {
```

Comment:
What if we make it even clearer: 

```suggestion
            if (inflight.callback) |callback| {
                assert(message.header.operation != .register);
                callback( ... );
            } else {
                assert(message.header.operation == .register);
            }
```